interface/patient_file/encounter/find_code_dynamic_ajax.php

```diff
@@ -294,8 +296,21 @@ function genFieldIdString($row)
         $out['aaData'][] = $arow;
     }
 } elseif ($what == 'codes') {
+    // for an external table with tons of codes for performance reasons we need to force them to search for something
+    $externalTableId = $code_types[$codetype]['external'] ?? 0;
+    /**
+     * @global $code_external_tables
+     */
+    $stopEmptySearch = $externalTableId && $code_external_tables[$externalTableId][SKIP_TOTAL_TABLE_COUNT] ?? false;
+    if (empty(trim($searchTerm)) && $stopEmptySearch) {
+        $out['iSearchEmptyError'] = xl('Search term is required for this code type.');
+        echo json_encode($out);
```

Comment:
if this is going to html (pretty sure it is), then need to surround the json_encode() with a text()

Reply:
It is pulled via AJAX by the DataTables  jquery library so it doesn't get directly injected into the page. Look at select_codes.php and you can see that its using xlj for all the translation screens.  If there's an escaping problem fundamentally in the DataTables library let me know and we can address it.